roles/postgresql/tasks/main.yml

```diff
@@ -3,76 +3,126 @@
   ansible.builtin.apt:
     pkg: ca-certificates
     state: present
+  when:
+    - ansible_os_family == "Debian"
 
 - name: PostgreSQL | Add PostgreSQL repository apt-key
   ansible.builtin.apt_key:
     url: "{{ postgres_apt_key_url }}"
     state: present
+  when:
+    - ansible_os_family == "Debian"
 
-- name: PostgreSQL | add postgres:10 appstream
+- name: PostgreSQL | add postgres:13 appstream
   ansible.builtin.dnf:
-    name: "@postgresql:10/client"
+    name: "postgresql"
     state: present
-    ignore_errors: true
   when:
     - running_on_server
-    - ansible_distribution == "CentOS"
+    - ansible_os_family == "RedHat"
 
 - name: PostgreSQL | add postgresql repository for latest ubuntu
   ansible.builtin.apt_repository:
     repo: 'deb http://apt.postgresql.org/pub/repos/apt/ {{ ansible_distribution_release }}-pgdg main'
     state: present
   when:
-    - ansible_distribution_release != "bionic"
-
-- name: PostgreSQL | add postgresql repository for Bionic
-  ansible.builtin.apt_repository:
-    repo: 'deb https://apt-archive.postgresql.org/pub/repos/apt/ {{ ansible_distribution_release }}-pgdg main'
-    state: present
-  when:
-    - ansible_distribution_release == "bionic"
+    - ansible_os_family == "Debian"
 
-- name: PostgreSQL | install postgresql server packages
+- name: PostgreSQL | install postgresql server packages (Ubuntu)
   ansible.builtin.apt:
     name: '{{ item }}'
     state: present
     cache_valid_time: '{{ apt_cache_timeout }}'
   loop:
     - "{{ postgres_packages }}"
-  when: postgresql_is_local
+  when:
+    - postgresql_is_local
+    - ansible_os_family == "Debian"
+
+- name: PostgreSQL | install postgresql server packages (RedHat)
+  ansible.builtin.dnf:
+    name: '{{ item }}'
+    state: present
+  loop:
+    - "{{ dnf_postgres_packages }}"
+  when:
+    - postgresql_is_local
+    - ansible_os_family == "RedHat"
 
-- name: PostgreSQL | install postgresql client packages
+- name: PostgreSQL | Ensure PostgreSQL is enabled and started
```

Comment:
I think to enable AND start it, we need `state: started` here as well as `enabled: true`.